src/asm2wasm.h

```diff
@@ -1537,6 +1537,7 @@ void Asm2WasmBuilder::optimize() {
   passRunner.add("remove-unused-names");
   passRunner.add("merge-blocks");
   passRunner.add("simplify-locals");
+  passRunner.add("constant-propagation");
```

Comment:
You are right about this pass not finding a lot of opportunities. My goal is to have a starting point, for more interesting data-flow type optimizations.